pkgs/development/compilers/openjdk/8.nix

```diff
@@ -223,8 +223,9 @@ let
       # Propagate the setJavaClassPath setup hook from the JRE so that
       # any package that depends on the JRE has $CLASSPATH set up
       # properly.
-      mkdir -p $jre/nix-support
+      mkdir -p {$out,$jre}/nix-support
       printWords ${setJavaClassPath} > $jre/nix-support/propagated-build-inputs
+      printWords ${setJavaClassPath} > $out/nix-support/propagated-build-inputs
```

Comment:
why not just 
`propagatedBuildInputs = [ setJavaClassPath ];`
[as it works in graal](https://github.com/NixOS/nixpkgs/blob/20b5f89384085535e7903152fd0d4e5562998837/pkgs/development/compilers/graalvm/enterprise-edition.nix#L94) ?

Reply:
You're absolutely right, that's even simpler! :+1: